src/bbtautau/postprocessing/postprocessing.py

```diff
@@ -971,6 +1014,65 @@ def bbtautau_assignment(
         sample.tt_mask = bbtt_masks["tt"]
 
 
+def _get_lepton_mask(sample, lepton_type: str, dR_cut: float) -> np.ndarray:
+    """
+    Calculates a boolean mask to select the highest-pT lepton of a given type
+    that is within a dR_cut of the ttFatJet.
+
+    Returns a boolean array with at most one 'True' per row (event).
+    """
+    # Use .to_numpy() for efficient computation.
+    # The jet's (N,) arrays are reshaped to (N, 1) to broadcast correctly
+    # against the lepton's (N, M) arrays, where M is the number of leptons.
+
+    lepton_eta = sample.get_var(f"{lepton_type}Eta")
+    lepton_phi = sample.get_var(f"{lepton_type}Phi")
+    jet_eta = sample.get_var("ttFatJetEta")[:, np.newaxis]
+    jet_phi = sample.get_var("ttFatJetPhi")[:, np.newaxis]
+
+    # print(sample.sample.label)
+    # print(np.sum(lepton_eta != PAD_VAL, axis=0)/len(lepton_eta))
+
+    # 1. Calculate dR for all leptons and create an initial mask
+    dR = np.sqrt((lepton_eta - jet_eta) ** 2 + (lepton_phi - jet_phi) ** 2)
```

Comment:
Subtracting phi directly can lead to edge effects since it's circular (e.g. phi1 = 2pi, phi2 = 0). See [here](https://github.com/scikit-hep/coffea/blob/a315da1fa307f1ec0d21c29e908e5b733603d7c0/src/coffea/nanoevents/methods/vector.py#L92-L97) for how to do it.